File: 2025-10-07 - Padroes de Projetos - Comparator/Exemplo_C#/SistemaAcademico.cs
using System;
using System.Collections.Generic;

public class SistemaAcademico
{
   public static void Main(string[] args)
   {
      List<Aluno> alunos = new List<Aluno>();

      alunos.Add(new Aluno("Carlos", 8.5f, 9.0f));
      alunos.Add(new Aluno("Ana", 9.2f, 8.8f));
      alunos.Add(new Aluno("Beatriz", 7.8f, 8.0f));

      Console.WriteLine("=== Lista Original ===");
      foreach (Aluno a in alunos)
      {
         Console.WriteLine(a);
      }

      alunos.Sort(new OrdenarPorNome());
      Console.WriteLine("\n=== Alunos ordenados por nome ===");
      foreach (Aluno a in alunos)
      {
         Console.WriteLine(a);
      }

      alunos.Sort(new OrdenarPorMedia());
      Console.WriteLine("\n=== Alunos ordenados por média (decrescente) ===");
      foreach (Aluno a in alunos)
      {
         Console.WriteLine(a);
      }
   }
}

File: 2025-10-07 - Padroes de Projetos - Comparator/Exemplo_Java/SistemaAcademico.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class SistemaAcademico {
   public static void main(String[] args) {
      List<Aluno> alunos = new ArrayList<>();

      alunos.add(new Aluno("Carlos", 8.5f, 9.0f));
      alunos.add(new Aluno("Ana", 9.2f, 8.8f));
      alunos.add(new Aluno("Beatriz", 7.8f, 8.0f));

      System.out.println("=== Lista Original ===");
      for (Aluno a : alunos) {
         System.out.println(a);
      }

      Collections.sort(alunos, new OrdenarPorNome());
      System.out.println("\n=== Alunos ordenados por nome ===");
      for (Aluno a : alunos) {
         System.out.println(a);
      }

      Collections.sort(alunos, new OrdenarPorMedia());
      System.out.println("\n=== Alunos ordenados por média (decrescente) ===");
      for (Aluno a : alunos) {
         System.out.println(a);
      }
   }
}
